From 7b3a5ed6652f8c227d7f4e9a4536ec5ee1310280 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Sun, 20 Sep 2020 18:34:13 +0000
Subject: [PATCH 04/11] send(): Cast to const char * for send()

Fixes

```c
../../modules/stream_out/rtcp.c: In function 'CloseRTCP':
../../modules/stream_out/rtcp.c:196:29: warning: pointer targets in passing argument 2 of 'send' differ in signedness [-Wpointer-sign]
  196 |     send (rtcp->handle, rtcp->payload, rtcp->length, 0);
      |                         ~~~~^~~~~~~~~
      |                             |
      |                             uint8_t * {aka unsigned char *}
In file included from ../../include/vlc_network.h:44,
                 from ../../modules/stream_out/rtcp.c:32:
D:/media-autobuild_suite/msys64/mingw32/i686-w64-mingw32/include/winsock2.h:1027:60: note: expected 'const char *' but argument is of type 'uint8_t *' {aka 'unsigned char *'}
 1027 |   WINSOCK_API_LINKAGE int WSAAPI send(SOCKET s,const char *buf,int len,int flags);
      |                                                ~~~~~~~~~~~~^~~
```

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 modules/stream_out/rtcp.c | 4 ++--
 src/network/rootbind.c    | 2 +-
 src/stream_output/sap.c   | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/modules/stream_out/rtcp.c b/modules/stream_out/rtcp.c
index d989efac44..a0e4892613 100644
--- a/modules/stream_out/rtcp.c
+++ b/modules/stream_out/rtcp.c
@@ -193,7 +193,7 @@ void CloseRTCP (rtcp_sender_t *rtcp)
 
     /* We are THE sender, so we are more important than anybody else, so
      * we can afford not to check bandwidth constraints here. */
-    send (rtcp->handle, rtcp->payload, rtcp->length, 0);
+    send (rtcp->handle, (const char *)rtcp->payload, rtcp->length, 0);
     net_Close (rtcp->handle);
     free (rtcp);
 }
@@ -227,6 +227,6 @@ void SendRTCP (rtcp_sender_t *restrict rtcp, const block_t *rtp)
     SetDWBE (ptr + 24, rtcp->bytes);
     memcpy (ptr + 28 + 4, rtp->p_buffer + 8, 4); /* SDES SSRC */
 
-    if (send (rtcp->handle, ptr, rtcp->length, 0) == (ssize_t)rtcp->length)
+    if (send (rtcp->handle, (const char *)ptr, rtcp->length, 0) == (ssize_t)rtcp->length)
         rtcp->counter = 0;
 }
diff --git a/src/network/rootbind.c b/src/network/rootbind.c
index f10c219559..0381b8d080 100644
--- a/src/network/rootbind.c
+++ b/src/network/rootbind.c
@@ -174,7 +174,7 @@ int rootwrap_bind (int family, int socktype, int protocol,
     memcpy (&ss, addr, (alen > sizeof (ss)) ? sizeof (ss) : alen);
 
     pthread_mutex_lock (&mutex);
-    if (send (sock, &ss, sizeof (ss), MSG_NOSIGNAL) != sizeof (ss))
+    if (send (sock, (const char *)&ss, sizeof (ss), MSG_NOSIGNAL) != sizeof (ss))
     {
         pthread_mutex_unlock (&mutex);
         return -1;
diff --git a/src/stream_output/sap.c b/src/stream_output/sap.c
index 0e699faeee..bfdd38e29e 100644
--- a/src/stream_output/sap.c
+++ b/src/stream_output/sap.c
@@ -132,7 +132,7 @@ static void *RunThread (void *self)
 
         vlc_list_foreach (p_session, &addr->sessions, node)
         {
-            send (addr->fd, p_session->data, p_session->length, 0);
+            send (addr->fd, (const char *)p_session->data, p_session->length, 0);
             deadline += vlc_tick_from_samples(addr->interval, addr->session_count);
 
             if (vlc_cond_timedwait(&addr->wait, &sap_mutex, deadline) == 0)
-- 
2.25.1

